trustmanager/x509memstore.go

```diff
@@ -84,8 +84,23 @@ func (s X509MemStore) RemoveCert(cert *x509.Certificate) error {
 	return nil
 }
 
+// RemoveAll removes all the certificates from the store
+func (s *X509MemStore) RemoveAll() error {
+
+	for _, cert := range s.fingerprintMap {
+		if err := s.RemoveCert(cert); err != nil {
+			return err
+		}
+	}
+
+	s.fingerprintMap = make(map[CertID]*x509.Certificate)
+	s.nameMap = make(map[string][]CertID)
```

Comment:
Seems like we either need these two lines or the loop above; is there some case where they're both needed?